app/src/parser.c

```diff
@@ -146,19 +146,25 @@ __Z_INLINE bool_t parser_isAmount(char *key) {
     return bool_false;
 }
 
-__Z_INLINE bool_t is_default_denom_base(const char *denom, uint8_t denom_len) {
-    if (tx_is_expert_mode()) {
-        return false;
+__Z_INLINE parser_error_t is_default_denom_base(const char *denom, uint8_t denom_len, bool *is_default) {
+    bool is_expert_or_default = false;
+    CHECK_PARSER_ERR(tx_is_expert_mode_or_not_default_chainid(&is_expert_or_default))
+    if (is_expert_or_default) {
+        *is_default = false;
+        return parser_ok;
     }
 
     if (strlen(COIN_DEFAULT_DENOM_BASE) != denom_len) {
-        return bool_false;
+        *is_default = bool_false;
```

Comment:
Can we replace this custom type and just use true/false? There are several parts that bool_t is used